src/ucp/core/ucp_context.c

```diff
@@ -851,6 +869,60 @@ static void ucp_fill_sockaddr_aux_tls_config(ucp_context_h context,
     }
 }
 
+static void ucp_fill_sockaddr_prio_list(ucp_context_h context,
+                                        const ucp_config_t *config)
+{
+    const char **sockaddr_tl_names = (const char**)config->sockaddr_cm_tls.cm_tls;
+    unsigned num_cfg_sockaddr_tls  = config->sockaddr_cm_tls.count;;
+    int                        idx = 0;
+    ucp_tl_resource_desc_t *resource;
+    ucp_rsc_index_t tl_id;
+    ucp_tl_md_t *tl_md;
+    int j;
+
+    /* Check if a list of sockadrr transports exists */
+    if (config->sockaddr_cm_tls.count == 0) {
+        ucs_debug("No sockaddr transports specified");
+        return;
+    }
+
+    /* Parse the sockadrr transports priority list */
+    for (j = 0; j < num_cfg_sockaddr_tls; j++) {
+        /* go over the proiroty list and find the transport's tl_id in the
+         * context's tl_bitmap. save the tl_id's for the client/server usage later */
+        if (strncmp(sockaddr_tl_names[j], "*", 1)) {
+            ucs_for_each_bit(tl_id, context->tl_bitmap) {
+                resource = &context->tl_rscs[tl_id];
+                tl_md    = &context->tl_mds[resource->md_index];
+                if (!strcmp(config->sockaddr_cm_tls.cm_tls[j],
```

Comment:
why need to compare again to '*'?

Reply:
it's not a comparison to '*'. 
it's comparing the sockaddr tl from the priority list to the resources in the context tl_bitmap.

Reply:
right